### PR TITLE
Wiki data

### PR DESCRIPTION
**Pull request by ablodge**
Hi Nathan,

I have fixed everything in our data and import-interface that I set out to fix. Can we set up a TODO list here for anything that needs to be done before merging with master and uploading our English data to Xposition? 

Thanks,
Austin

@nschneid

### CODE REVIEW

**Comment by nschneid** [comment]
Awesome. Key remaining items:

- [ ] Sentence and token views (#46, #61)—if you set up the URLs and point them to templates I can fill in the templates
- [ ] Corpus data updates (#69, #70)
- [ ] #58

**Comment by ablodge** [comment]
I have everything working on my end. Also, automated import for supersense and construal articles works and looks great.

**Comment by nschneid** [comment]
Beautiful! Since I'm not falling asleep, will spend some time perusing the pages and looking for LaTeX conversion bugs.

**Comment by ablodge** [comment]
I had to change the url pattern for corpus_sentences to `en/corpus/#/sent_id` to avoid colision with the usage url pattern.

commit: d19a2f1

File: scripts/tex2markdown.py [hunk 1]
```diff
@@ -328,6 +321,22 @@ def convert(ifile, ofile, title):
...
 
+        # backspaces
+        data = data.replace("\\$", "$")
+        data = data.replace("\\#", "#")
+        data = data.replace("\\%", "%")
+        # data = data.replace(r"\\{}", " ")
+        data = data.replace(r'\{', '{')
+        data = data.replace(r'\}', '}')
+        data = data.replace("\\\\", " ")
+        data = data.replace("\\ ", "")
```
Comment (nschneid) [inline scripts/tex2markdown.py]: Replacement should be `" "`, I think. :)

File: scripts/tex2markdown.py [hunk 2]
```diff
@@ -328,6 +321,22 @@ def convert(ifile, ofile, title):
...
+        data = data.replace("etc.\\", "etc.")
+        data = data.replace("etc.)\\", "etc.)")
```
Comment (nschneid) [inline scripts/tex2markdown.py]: The above two cases should be covered by the previous one (backslash+space).

Comment (ablodge) [reply scripts/tex2markdown.py]: 'etc.\\' and 'etc.)\\' do show up though. They might have a line break instead of a space.

Comment (ablodge) [reply scripts/tex2markdown.py]: Yes! That is true.

**Comment by ablodge** [comment]
Do we want fancy quotes: `“`, `‘`? To be consistent I can get rid of them.

**Comment by nschneid** [comment]
Fancy quotes are nice. :) Since they're specified in the LaTeX I think we may as well keep them.

File: wiki/plugins/metadata/urls.py [hunk 3]
```diff
@@ -9,7 +9,7 @@ def get_root_urls(self):
         # customize certain root URLs to override putting them under _plugin/metadata
         root_url_patterns = [
             url(r'^ex/(?P<exnum>\d+)/$', views.PTokenView.as_view(), name='ptoken_view'), #  /ex/3495/
-            url('^(?P<lang>[a-z][a-z](-[a-z]+)?)/(?P<corpus>[^/]+)/#/(?P<sent_id>[^/]+)/$', views.CorpusSentenceView.as_view(), name='corpus_sentence_view'), #  /en/corpus/streusle4.1/reviews-001325-0003
+            url('^(?P<lang>[a-z][a-z](-[a-z]+)?)/(?P<corpus>[^/]+)/~/(?P<sent_id>[^/]+)/$', views.CorpusSentenceView.as_view(), name='corpus_sentence_view'), #  /en/corpus/streusle4.1/reviews-001325-0003
```
Comment (ablodge) [inline wiki/plugins/metadata/urls.py]: I'm sure we can think of something better. This is to avoid url collision with the pattern /en/prep/construal.

**Comment by nschneid** [comment]
One way would be to customize the regex for corpora to require the corpus name to contain a digit, and prevent adposition names from containing digits. I think we can require that all corpus names include a version number. Do we ever expect adpositions with digits in the canonical spelling? Digits are used for tones in Pinyin, and for some letters in Arabizi, but I think the canonical form should use the native script anyway.

If we go with this constraint it should be incorporated into the form field validation for corpora and adpositions.

**Comment by ablodge** [comment]
TODO list:
[ ] fix CorpusSentence url 
[ ] Other Latex fixes?
[ ] Other style fixes?

**Comment by ablodge** [comment]
Okay! Everything works. We might plan some more testing, but I see no reason why we can't merge with master.

**Pull request merged**